### ISSUE TITLE
Sonic Origins

### ISSUE THREAD

**Issue by MagmanPlay** [strato-emu/compat-list#696]
### Tested Build

Skyline Edge 16

### Android Version

12

### SoC type

Snapdragon

### Device GSMArena link

https://m.gsmarena.com/xiaomi_poco_x3_pro-10802.php

### Game Behaviour

Game plays very poorly, Menu bugs are low fps in the game but mostly everything works, there is room to grow

### Is Log Level is set to 'Info'?

- [X] Yes

### Log file
 
Can't load

Screenshots
![Screenshot_2022-12-08-22-08-53-156_skyline emu](https://user-images.githubusercontent.com/95186700/206547310-bd9700a0-d92e-42a4-a51b-cd20a7fccd17.jpg)
![Screenshot_2022-12-08-22-02-29-582_skyline emu](https://user-images.githubusercontent.com/95186700/206547315-560b5542-12e4-44f1-bf21-dcdcd8cf17af.jpg)
![Screenshot_2022-12-08-22-07-51-596_skyline emu](https://user-images.githubusercontent.com/95186700/206547318-704b715a-5215-4d56-90be-f8ae6d2654dc.jpg)
![Screenshot_2022-12-08-22-05-55-544_skyline emu](https://user-images.githubusercontent.com/95186700/206547319-c40c7f9f-73ac-4e4e-8bb6-f23fafe4d335.jpg)
![Screenshot_2022-12-08-22-04-28-710_skyline emu](https://user-images.githubusercontent.com/95186700/206547324-88476331-0afc-46ed-ac06-7174bb05fe06.jpg)

**Comment by Ghjkyfgifu** [comment]
Jxohcpj

**Comment by marcemira** [comment]
https://discord.com/channels/545842171459272705/986020197146169354/1016383524992991264
[emulation.sklog.zip](https://github.com/skyline-emu/skyline-games-list/files/9500467/emulation.sklog.zip)

**Comment by daniilstrunin** [comment]
Yes, there is progress, I managed to get the same screen as you have, the FPS counter shows 60. I'm waiting until it becomes fully playable

**Comment by daniilstrunin** [comment]
New Test
Android 12
Skyline Edge 14
Poco X3 Pro
Game show all retro engine menu , when the HedgehogEngine menu loads, everything freezes
video bellow

https://user-images.githubusercontent.com/95186700/204266861-28b8df0b-2935-4463-a5f3-d22efcb878a2.mp4
Only need to fix HedgehogEngine on Origins,Frontiers and OK👌

**Comment by daniilstrunin** [comment]
New test in first post! Check It!

**Comment by DeltaVRG** [comment]
On version 2147 game playable but BIG graphic problems, a lot of games don't work
